### PR TITLE
Jail filters rewrite + quick bail/bust + fast refresh

### LINKED ISSUE #71: Quick Bust and Quick Bail
- [ ] Settings @ Jail = Quick Bail and Bust
- [ ] Add Quick bail and Quick Bust.

### PR DESCRIPTION
**Pull request by Harpush**
Closes #71 .
image
image
image
image

### CODE REVIEW

**Review by DeKleineKobini — COMMENTED** [review]
Will review more when I have some more time.

File: extension/scripts/features/jailfilter/ttJailFilter.js [hunk 1]
```diff
@@ -1,275 +1,595 @@
...
 		{
 			storage: ["settings.pages.jail.filter"],
 		},
 		null
 	);
 
-	function initialiseFilters() {
-		CUSTOM_LISTENERS[EVENT_CHANNELS.JAIL_SWITCH_PAGE].push(filtering);
-	}
+	const storageFilters = filters;
```
[the remark this answers is not in the document]
Comment (Harpush) [reply extension/scripts/features/jailfilter/ttJailFilter.js:17]: There are places where I create a local variable called `filters` so wanted to make sure it won't get mixed up when reading. That way it's easy to tell apart between a storage based and not storage based. IMO that actual solution should be to use storage globally as `STORAGE.filters` for example.

File: extension/scripts/features/jailfilter/ttJailFilter.js [hunk 2]
```diff
@@ -1,275 +1,595 @@
...
 
 (async () => {
 	featureManager.registerFeature(
 		"Jail Filter",
 		"jail",
 		() => settings.pages.jail.filter,
-		initialiseFilters,
-		addFilters,
-		removeFilters,
+		() => {},
```
[the remark this answers is not in the document]
Comment (Harpush) [reply extension/scripts/features/jailfilter/ttJailFilter.js]: Fixed

**Comment by DeKleineKobini** [comment]
Needs some improvement on mobile view.

Proposed changes for mobile design (`body.tt-mobile` can achieve that with css only): 
- `flex-direction: column; ` on the filter content
- some extra space between the sections

**Review by DeKleineKobini — COMMENTED** [review]
Refresh buttons do not appear for me under any circumstance.

When showing them manually: why are there 2 refresh buttons in the user list position?

**Comment by Harpush** [comment]
> Refresh buttons do not appear for me under any circumstance.
> 
> When showing them manually: why are there 2 refresh buttons in the user list position?

Thats weird... Any way I can reproduce it?
There are two for quick bust and bail (each in the buttons place) and appearing based on whether you have quick enabled or not

File: extension/scripts/global/globalData.js [hunk 3]
```diff
@@ -465,6 +498,11 @@ const CONTRIBUTORS = {
 		name: "Tesa",
 		color: "brown",
 	},
+	AllMight: {
```
[the remark this answers is not in the document]
Comment (Harpush) [reply extension/scripts/global/globalData.js:501]: 👍

File: extension/scripts/global/globalData.js [hunk 4]
```diff
@@ -134,6 +134,32 @@ const ttStorage = new (class {
...
+	unknownFactions: "__unknown__",
+	// Quick modes
+	bust: "bust",
+	bail: "bail",
+	// Time
+	timeMin: 0,
+	timeMax: 100,
+	timeStep: 1,
+	// Level
+	levelMin: 0,
```
[the remark this answers is not in the document]
Comment (Harpush) [reply extension/scripts/global/globalData.js]: Done

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**